### PR TITLE
Jlib tests and cleaning

### PR DESCRIPTION
**Pull request by michalwski**
This PR creates tests for jlib module and cleans the module code a bit.

### CODE REVIEW

**Comment by ppikula** [comment]
Great code man!  :+1: 

I see that you started testing the jid functions. What about moving them all to a
separate module - for instance `jid.erl` (vim-wrangler to the rescue ;))?

 By doing that we could be more "ecological"(use less letters, save columns) - `jlib:jid_to_binary`  -> `jid:to_binary`, `jlib:make_jid` -> `jlib:new/make` etc.

**Comment by michalwski** [comment]
Also we can merge functions `jid_tolower/1` and `jid_to_lower/1` as they do the same and both are used from other modules... I think introducing `jid.erl` module should be done in a separate PR.

**Comment by pzel** [comment]
Code reviewed by pair programming.

**Comment by michalwski** [comment]
Thanks @pzel. Please merge this PR if you don't have further comments.

File: apps/ejabberd/src/jlib.erl [hunk 1]
```diff
@@ -368,22 +363,14 @@ jid_to_binary({Node, Server, Resource}) ->
          end,
     S3.
 
--spec is_nodename([] | binary()) -> boolean().
+-spec is_nodename(<<>> | binary()) -> boolean().
 is_nodename([]) ->
```
Comment (fenek) [inline apps/ejabberd/src/jlib.erl]: This clause should be removed if we are sure this function won't be called for lists.

Comment (michalwski) [inline apps/ejabberd/src/jlib.erl]: I changed this to empty binary. Empty list was a leftover after switching form lists to binary. This line was not covered by our tests and that's why we didn't catch it before.

**Pull request merged**